gammapy/makers/background/tests/test_fov.py

```diff
@@ -284,3 +290,20 @@ def test_fov_bkg_maker_scale_fail(obs_dataset, exclusion_mask, caplog):
     assert "WARNING" in [_.levelname for _ in caplog.records]
     message1 = "FoVBackgroundMaker failed. Only -1940 background counts outside exclusion mask for test-fov. Setting mask to False."
     assert message1 in [_.message for _ in caplog.records]
+
+
+@requires_data()
+def test_fov_bkg_maker_mask_fit_handling(obs_dataset, exclusion_mask):
+    fov_bkg_maker = FoVBackgroundMaker(method="scale", exclusion_mask=exclusion_mask)
+    test_dataset = obs_dataset.copy(name="test-fov")
+    region = CircleSkyRegion(obs_dataset._geom.center_skydir, Angle(0.4, "deg"))
+    mask_fit = obs_dataset._geom.region_mask(regions=[region])
+    test_dataset.mask_fit = mask_fit
+
+    dataset = fov_bkg_maker.run(test_dataset)
+    assert np.all((test_dataset.mask_fit == mask_fit).data) == True
```

Comment:
`assert_allclose` should better for this

Reply:
I think for booleans there should be no uncertainty. However one should simplify the expression to:
```
assert np.all(test_dataset.mask_fit == mask_fit)
```
`Map` objects are interpreted as `np.ndarray` when put into a numpy function...

Reply:
Can you try this solution @luca-giunti ?

Reply:
@registerrier Sure, sorry for the delay